fprime-vxworks/VxWorks/VxWorks7/WatchdogTimer.cpp

```diff
@@ -0,0 +1,69 @@
+#include <Os/WatchdogTimer.hpp>
+#include <Fw/Types/Assert.hpp>
+
+#include <vxWorks.h>
+#include <wdLib.h>
+#include <sysLib.h>
+
+namespace Os {
+
+    //!< This wrapper is needed due to the change in FUNCPTR's signature
+    //!< from VxWorks6 to VxWorks7. This static function implements the
+    //!< FUNCPTR signature required by VxWorks7 and calls the routine
+    //!< with the expected parameter. This wrapper function is expected to be
+    //!< passed in to wdStart.
+    static int myWatchdogCallback(
+        int arg //!< arg is the WatchdogTimer instance that initiated the wdStart
+    )
+    {
+        WatchdogTimer *comp = reinterpret_cast<WatchdogTimer*>(arg);
+        FW_ASSERT(comp != nullptr);
+        WatchdogTimer::WatchdogCb cb = comp->getCallback();
+        cb(comp->getParameter());
+        return OK;
+    }
```

Comment:
@LeStarch , take a look at this piece of code. I added this wrapper to get around the different FUNCPTR definitions between Vx6 and Vx7. This wrapper needs access to some private member variables, hence the https://github.com/nasa/fprime/pull/1649 PR.

Reply:
I see. Since callback and parameter aren't natively passable to the Os watchdog code, then you have to wrap it.  However,  it seems to me:

```
void expire() {
    this->callback(this->parameter);
}
```

Should be wrapped in a function such that all your above code has to do is call expire.  That way the implementation doesn't require exposing private members like getter/setter.